src/omv/imlib/simd.h

```diff
@@ -0,0 +1,1725 @@
+    return (v128_t) {
+        .u32 = { x * 0x01010101 }
+    };
+    #endif
+}
+
+// GCC does not vectorize assignment from a scalar to a vector.
+static inline v128_t vdup_s8(int8_t x) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vdupq_n_s8(x);
+    #else
+    return (v128_t) {
+        .s32 = { (x & 0xFF) * 0x01010101 }
+    };
+    #endif
+}
+
+// GCC does not vectorize assignment from a scalar to a vector.
+static inline v128_t vdup_u16(uint16_t x) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vdupq_n_u16(x);
+    #else
+    return (v128_t) {
+        .u32 = { x * 0x00010001 }
+    };
+    #endif
+}
+
+// GCC does not vectorize assignment from a scalar to a vector.
+static inline v128_t vdup_s16(int16_t x) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vdupq_n_s16(x);
+    #else
+    return (v128_t) {
+        .s32 = { (x & 0xFFFF) * 0x00010001 }
+    };
+    #endif
+}
+
+static inline v128_t vdup_u32(uint32_t x) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vdupq_n_u32(x);
+    #else
+    return (v128_t) {
+        .u32 = { x }
+    };
+    #endif
+}
+
+static inline v128_t vdup_s32(int32_t x) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vdupq_n_s32(x);
+    #else
+    return (v128_t) {
+        .s32 = { x }
+    };
+    #endif
+}
+
+static inline v128_t vidup_u8(uint32_t start, uint32_t increment) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vidupq_n_u8(start, increment);
+    #else
+    v128_t r;
+    r.u8[0] = start;
+    r.u8[1] = start + increment;
+    r.u8[2] = start + (increment * 2);
+    r.u8[3] = start + (increment * 3);
+    return r;
+    #endif
+}
+
+static inline v128_t vidup_u16(uint32_t start, uint32_t increment) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vidupq_n_u16(start, increment);
+    #else
+    v128_t r;
+    r.u16[0] = start;
+    r.u16[1] = start + increment;
+    return r;
+    #endif
+}
+
+static inline v128_t vidup_u32(uint32_t start, uint32_t increment) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vidupq_n_u32(start, increment);
+    #else
+    return (v128_t) {
+        .u32 = { start }
+    };
+    #endif
+}
+
+static inline v128_t vidup_u32_unaligned(uint32_t start, uint32_t increment) {
+    #if (__ARM_ARCH >= 8)
+    #ifdef NDEBUG
+    v128_t offsets = (v128_t) vidupq_n_u32(start, increment);
+    #else
+    v128_t offsets;
+    offsets.u32[0] = start;
+    offsets.u32[1] = start + increment;
+    offsets.u32[2] = start + (increment * 2);
+    offsets.u32[3] = start + (increment * 3);
+    #endif
+    offsets.u32 = vsliq_n_u32(offsets.u32, offsets.u32, 8);
+    offsets.u32 = vsliq_n_u32(offsets.u32, offsets.u32, 16);
+    offsets.u8 = vaddq(offsets.u8, viwdupq_n_u8(0, 4, 1));
+    return offsets;
+    #else
+    return (v128_t) {
+        .u32 = { start }
+    };
+    #endif
+}
+
+static inline v128_t vrol(v128_t v0, uint32_t n) {
+    #if (__ARM_ARCH >= 8)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+    uint32_t c;
```

Comment:
```suggestion
    uint32_t c = 0;
```
Is `c` causing the issue ?

Reply:
Yeah, I was trying to avoid an ASM op assigning it to zero. However, the compiler assigns it to zero in the ASM. Will modify this and remove the diagnostic.